src/hash.cpp

```diff
@@ -297,8 +297,12 @@ const std::string &Hash::digest()
 
   m_value.resize(multihash.size() * 2);
 
-  for(size_t i = 0; i < multihash.size(); ++i)
-    sprintf(&m_value[i * 2], "%02x", multihash[i]);
+  for(size_t i = 0; i < multihash.size(); ++i) {
+      char buf[3];
+      snprintf(buf, sizeof(buf), "%02x", multihash[i]);
+      m_value[i * 2] = buf[0];
+      m_value[i * 2 + 1] = buf[1];
```

Comment:
There's no point in writing to a temporary buffer before copying to the final string. The direct sprintf to snprintf conversion would be:

```suggestion
      snprintf(&m_value[i * 2], 3, "%02x", multihash[i]);
```

...though there's also no point in writing a temporary null terminator every 2 digits only to overwrite it with the next digit in the next iteration, so this loop would be better written as:

```diff
diff --git c/src/hash.cpp i/src/hash.cpp
index dbe1fc3..e47a01d 100644
--- c/src/hash.cpp
+++ i/src/hash.cpp
@@ -297,8 +297,12 @@ const std::string &Hash::digest()
 
   m_value.resize(multihash.size() * 2);
 
-  for(size_t i = 0; i < multihash.size(); ++i)
-    sprintf(&m_value[i * 2], "%02x", multihash[i]);
+  static const char hex[] = "0123456789abcdef";
+
+  for(size_t i = 0; i < multihash.size(); ++i) {
+    m_value[i * 2    ] = hex[multihash[i] >> 4];
+    m_value[i * 2 + 1] = hex[multihash[i] & 0xf];
+  }
 
   return m_value;
 }
```